### PR TITLE
Enable checking for provided repositories, whether or not they are enabled

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

File: crowbar_framework/lib/crowbar/repository.rb [hunk 1]
```diff
@@ -156,6 +156,8 @@ def provided_and_enabled?(feature, platform = nil, arch = nil)
             next unless provided_features.include? feature
             found = true
 
+            next unless check_enabled
```
Comment (vuntz) [inline crowbar_framework/lib/crowbar/repository.rb]: I think you can `break`.

File: crowbar_framework/lib/crowbar/repository.rb [hunk 2]
```diff
@@ -129,19 +129,19 @@ def check_all_repos
         repochecks
       end
 
-      def provided_and_enabled?(feature, platform = nil, arch = nil)
+      def provided_and_enabled?(feature, platform = nil, arch = nil, check_enabled = true)
```
Comment (vuntz) [inline crowbar_framework/lib/crowbar/repository.rb]: I'm somehow a bit uneasy with the fact that we allow disabling a feature of this method that is in the name of the method. I don't know if that's okay or not from a style perspective. (An alternative would be to move the content of this method to a helper and call the helper from here).

**Comment by tboerger** [comment]
:+1:

**Comment by jsuchome** [comment]
@vuntz better this way?

**Comment by vuntz** [comment]
> @vuntz better this way?

Yes! I would just make the helper method private or protected (so just move it down in a later commit).

**Comment by vuntz** [comment]
Thanks for the changes, +1

File: crowbar_framework/lib/crowbar/repository.rb [hunk 3]
```diff
@@ -130,18 +130,59 @@ def check_all_repos
...
+
+      private
+
+      def provided_with_enabled?(feature, platform = nil, arch = nil, check_enabled = true)
         answer = false
 
         if platform.nil?
           all_platforms.each do |p|
-            if provided_and_enabled?(feature, p)
+            if provided_and_enabled?(feature, p, arch, check_enabled)
```
Comment (nkrinner) [inline crowbar_framework/lib/crowbar/repository.rb]: Maybe I am overlooking something here, but this calls the provided_and_enabled? method which does nothing but calling this method  (provided_with_enabled?) again, and also has the wrong number of arguments?

File: crowbar_framework/lib/crowbar/repository.rb [hunk 4]
```diff
@@ -130,18 +130,59 @@ def check_all_repos
...
               answer = true
               break
             end
           end
         elsif arch.nil?
           arches(platform).each do |a|
-            if provided_and_enabled?(feature, platform, a)
+            if provided_and_enabled?(feature, platform, a, check_enabled)
```
Comment (nkrinner) [inline crowbar_framework/lib/crowbar/repository.rb]: Same here

**Comment by nkrinner** [comment]
+1

**Comment by vuntz** [comment]
+1

**Pull request merged**